### PR TITLE
Extractor error to include extractor type/name

### PR DESCRIPTION
**Pull request by nonibansal**

### CODE REVIEW

File: fennel/internal_lib/to_proto/to_proto.py [hunk 1]
```diff
@@ -473,12 +478,12 @@ def _extractor_to_proto(
                 inputs.append(feature_to_proto_as_input(f))
         elif isinstance(input, Featureset):
             raise TypeError(
-                f"Extractor input {input} is a Featureset, please use a"
+                f"{extractor_name} Extractor input {input} is a Featureset, please use a"
```
Comment (nikhilgarg28) [inline fennel/internal_lib/to_proto/to_proto.py:481]: nit: now that 'Extractor' isn't the first letter of the sentence, make it start with small case. Also, use small case for 'Featureset' & DataFrame later.

File: fennel/internal_lib/to_proto/to_proto.py [hunk 2]
```diff
@@ -473,12 +478,12 @@ def _extractor_to_proto(
...
                 f"DataFrame of features"
             )
         else:
             raise TypeError(
-                f"Extractor input {input} is not a Feature or "
+                f"{extractor_name} Extractor input {input} is not a Feature or "
```
Comment (nikhilgarg28) [inline fennel/internal_lib/to_proto/to_proto.py:486]: Same here.

**Review by nikhilgarg28 — CHANGES_REQUESTED** [review]

File: fennel/internal_lib/to_proto/to_proto.py (hunk 1 above)
Comment (nonibansal) [reply fennel/internal_lib/to_proto/to_proto.py]: changed

File: fennel/internal_lib/to_proto/to_proto.py (hunk 2 above)
Comment (nonibansal) [reply fennel/internal_lib/to_proto/to_proto.py]: changed

File: fennel/internal_lib/to_proto/to_proto.py [hunk 3]
```diff
@@ -464,6 +464,11 @@ def feature_to_proto_as_input(f: Feature) -> fs_proto.Input:
 def _extractor_to_proto(
     extractor: Extractor, fs: Featureset, fs_obj_map: Dict[str, Featureset]
 ) -> fs_proto.Extractor:
+    extractor_name = (
+        extractor.name
+        if extractor.extractor_type == ExtractorType.PY_FUNC
+        else "Auto Generated"
+    )
```
Comment (aditya-nambiar) [inline fennel/internal_lib/to_proto/to_proto.py:471]: dont autogenerated extractors not have a name ? _fennel_alias .. or _fennel_lookup ?

File: fennel/internal_lib/to_proto/to_proto.py [hunk 4]
```diff
@@ -473,13 +478,13 @@ def _extractor_to_proto(
                 inputs.append(feature_to_proto_as_input(f))
         elif isinstance(input, Featureset):
             raise TypeError(
-                f"Extractor input {input} is a Featureset, please use a"
-                f"DataFrame of features"
+                f"{extractor_name} extractor input {input} is a Featureset, please use a"
+                f"dataFrame of features"
```
Comment (aditya-nambiar) [inline fennel/internal_lib/to_proto/to_proto.py:482]: s/dataFrame/dataframe of feature values

File: fennel/internal_lib/to_proto/to_proto.py [hunk 5]
```diff
@@ -473,13 +478,13 @@ def _extractor_to_proto(
...
             )
         else:
             raise TypeError(
-                f"Extractor input {input} is not a Feature or "
-                f"a DataFrame of features, but a {type(input)}"
+                f"{extractor_name} extractor input {input} is not a feature or "
+                f"a dataFrame of features, but a {type(input)}"
```
Comment (aditya-nambiar) [inline fennel/internal_lib/to_proto/to_proto.py:487]: s/dataFrame/dataframe
but of type {type(input)}

File: fennel/internal_lib/to_proto/to_proto.py (hunk 3 above)
Comment (nonibansal) [reply fennel/internal_lib/to_proto/to_proto.py:471]: yup they do have, moving the error ton that.

File: fennel/internal_lib/to_proto/to_proto.py (hunk 4 above)
Comment (nonibansal) [reply fennel/internal_lib/to_proto/to_proto.py:482]: done

File: fennel/internal_lib/to_proto/to_proto.py (hunk 5 above)
Comment (nonibansal) [reply fennel/internal_lib/to_proto/to_proto.py:487]: done.

File: fennel/internal_lib/to_proto/to_proto.py [hunk 6]
```diff
@@ -485,13 +485,13 @@ def _extractor_to_proto(
                 inputs.append(feature_to_proto_as_input(f))
         elif isinstance(input, Featureset):
             raise TypeError(
-                f"Extractor input {input} is a Featureset, please use a"
-                f"DataFrame of features"
+                f"{extractor.name} extractor input {input} is a Featureset, please use a"
```
Comment (aditya-nambiar) [inline fennel/internal_lib/to_proto/to_proto.py:488]: Can we use ` to indicate its a user input eg
`{extractor.name}`

**Review by aditya-nambiar — APPROVED** [review]

File: fennel/internal_lib/to_proto/to_proto.py (hunk 6 above)
Comment (nonibansal) [reply fennel/internal_lib/to_proto/to_proto.py]: added

**Pull request merged**